Shared/Samples/Find route around barriers/FindRouteAroundBarriersView.Model.swift

```diff
@@ -0,0 +1,207 @@
+        
+        /// The graphics overlay for the stop graphics.
+        private let stopGraphicsOverlay = GraphicsOverlay()
+        
+        /// The graphics overlay for the barrier graphics.
+        private let barrierGraphicsOverlay = GraphicsOverlay()
+        
+        /// The graphics overlay for the route graphics.
+        private let routeGraphicsOverlay = GraphicsOverlay()
+        
+        /// The blue marker symbol for the stop graphics.
+        private let stopSymbol = {
+            let markerImage = UIImage(named: "BlueMarker")!
+            let markerSymbol = PictureMarkerSymbol(image: markerImage)
+            markerSymbol.offsetY = markerImage.size.height / 2
+            return markerSymbol
+        }()
+        
+        /// The graphic for the route.
+        private let routeGraphic = Graphic(
+            symbol: SimpleLineSymbol(style: .solid, color: .yellow, width: 5)
+        )
+        
+        /// The graphic for the direction route.
+        let directionGraphic = Graphic(
+            symbol: SimpleLineSymbol(style: .dashDot, color: .orange, width: 5)
+        )
+        
+        /// The route task for routing.
+        let routeTask = RouteTask(url: .sanDiegoNetworkAnalysis)
+        
+        /// The route parameters for routing with the route task.
+        var routeParameters: RouteParameters?
+        
+        /// The resulting route from a routing operation with the route task.
+        private(set) var route: Route?
+        
+        /// The text with the time and distance of the current route.
+        @Published private(set) var routeInfoText = ""
+        
+        /// The count of stops currently on the map.
+        @Published private(set) var stopsCount: Int = 0
+        
+        init() {
+            routeGraphicsOverlay.addGraphics([routeGraphic, directionGraphic])
+            graphicsOverlays = [routeGraphicsOverlay, barrierGraphicsOverlay, stopGraphicsOverlay]
+            updateRouteInfoText()
+        }
+        
+        // MARK: Methods
+        
+        /// Adds a stop graphic to the map.
+        /// - Parameter point: The point to add the stop graphic at.
+        func addStopGraphic(at point: Point) {
+            // Create a text symbol with the next index.
+            let textSymbol = TextSymbol(
+                text: "\(stopsCount + 1)",
+                color: .white,
+                size: 20,
+                horizontalAlignment: .center,
+                verticalAlignment: .middle
+            )
+            textSymbol.offsetY = stopSymbol.offsetY
+            
+            // Create a graphic with the marker symbol and text symbol.
+            let compositeSymbol = CompositeSymbol(symbols: [stopSymbol, textSymbol])
+            let stopGraphic = Graphic(geometry: point, symbol: compositeSymbol)
+            
+            // Add the new graphic to the stop graphics overlay.
+            stopGraphicsOverlay.addGraphic(stopGraphic)
+            stopsCount += 1
```

Comment:
Why do you need a stop count? Can't you just check how many elements are in `stopGraphicsOverlay` ?

Reply:
No. The "Route" button's `disabled` modifier won't update just by watching `stopGraphicsOverlay.graphics.count` and requires some type of State variable to work correctly.